core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -664,6 +675,13 @@ private void doStop(CompletableFuture<Void> future,
             });
         }
 
+        private void unscheduleExceptionReporting() {
+            final ServerErrorHandler serverErrorHandler = config().errorHandler();
+            if (serverErrorHandler instanceof ExceptionReportingServerErrorHandler) {
+                ((ExceptionReportingServerErrorHandler) serverErrorHandler).stop();
+            }
```

Comment:
This type checking looks like a hacky way that could be broken if #3421 is added.
Should we use `ServerListener` API to get the lifestyle of a `Server`? 
```java
class ExceptionReportingServerErrorHandler implements ServerErrorHandler, ServerListener {
    @Override
    public void serverStarting(Server) {
        reportingTaskFuture = workerGroup.scheduleAtFixedRate(...);
        ...
    }

    @Override
    public void serverStopping(Server server) {
        ...
        reportingTaskFuture.cancel(true);
        ...
    }
}
```
`ExceptionReportingServerErrorHandler` can be added to https://github.com/ikhoon/armeria/blob/86cb7cce3bebe5c218f48d21184ef0ab51512feb/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L174

Reply:
This looks much more clean. Thank you for 👍 👍 👍  the comment.

Reply:
I'll remove `scheduleExceptionReporting` and `unscheduleExceptionReporting` methods and ˆmplement the `ServerListener` instead.